src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsProjectRestoreInfo.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Contains project metadata needed for project restore operation
+    /// </summary>
+    [ComImport]
+    [Guid("ab43992d-b977-436d-84c2-e76aeed20de2")]
+    public interface IVsProjectRestoreInfo
+    {
+        /// <summary>
+        /// Base intermediate path of the project
+        /// </summary>
+        string BaseIntermediatePath { get; }
+
+        /// <summary>
+        /// Target frameworks
+        /// </summary>
+        IEnumerable<IVsTargetFrameworkInfo> Frameworks { get; }
```

Comment:
Why not guarantee it?
Also, could we make this a IReadOnlyList to allow indexing? I am a little concerned passing `IEnumerable` over component boundaries since it could do non-trivial work per enumeration.